en/editor/publish/build-open-data-context.md

```diff
@@ -0,0 +1,113 @@
+# Building Open Data Context Project
+
+## Introduction to Open Data Context
+
+At present, **WeChat**, **Baidu** and **ByteDance** are small game platforms that add the concept of **Open Data Context**, which is a separate game execution environment, in order to protect their social relationship chain data. The resources, engines, and applications in the Open Data Context are completely isolated from the main game, and only in the Open Data Context can developers access the relationship chain data through the open interface provided by the platform to implement some features such as ranked version.
+
+In Cocos Creator 3D, we have deprecated the maintenance of the Canvas Renderer module. The Open Data Context now only supports Canvas rendering.
+In the previous scheme, we used Cocos Creator 2D to build Open Data Context projects. As of v3.0, we've deprecated this practice and provided with a built-in Canvas engine. The engine is provided by the WeChat team and is a front-end lightweight Canvas engine based on XML+CSS design.
+
+And with a few basic front-end skills, a developer can simply implement a ranked version of the feature.
+
+## Building processes
+
+1. Add the `SubContextView` component to nodes that need to display Open Data Context in the project.
+
+2. The platforms that currently support building Open Data Context projects are **WeChat**, **Baidu** and **ByteDance**. In the build options of these platforms, check **Generate Open Data Context Template**, click **Recompile**.
+
+    ![](./build-open-data-context/generate-template.png)
+
+3. After build, a project named **openDataContext** will be generated in the directory, which is a built-in template project in Cocos Creator.  
+**Note**: Next time you build, if the **openDataContext** directory exists in the build directory, it will not be overwritten.
+
+    ![](./build-open-data-context/build-output.png)
+
+4. Use the developer tool provided by the platform to open the built main project, you can see the effect of the Open Data Context Template Project.
+
+    ![](./build-open-data-context/show-in-devtool.png)
+
+## Customization on Open Data Context Project
+
+Before we can prepare for custom Open Data Context projects, we need to understand some basic skills, which are included here [minigame-canvas-engine quick start](https://wechat-miniprogram.github.io/minigame-canvas-engine/api/guide.html#%E5%AE%89%E8%A3%85) and [doT template engine use](http://olado.github.io/doT/?spm=a2c6h.12873639.0.0.36f45227oKu0XO)
```

Comment:
Before we can prepare for custom Open Data Context projects, we need to understand some basic skills, which are included here [minigame-canvas-engine quick start](https://wechat-miniprogram.github.io/minigame-canvas-engine/api/guide.html#%E5%AE%89%E8%A3%85) and [doT template engine use](http://olado.github.io/doT/?spm=a2c6h.12873639.0.0.36f45227oKu0XO)
-->
Before starting a custom **Open Data Context Project**, it is important to understand some basic skills:

- [minigame-canvas-engine quick start](https://wechat-miniprogram.github.io/minigame-canvas-engine/api/guide.html#%E5%AE%89%E8%A3%85)
- [doT template engine use](http://olado.github.io/doT/?spm=a2c6h.12873639.0.0.36f45227oKu0XO)